cpp/src/decisiontree/batched-levelalgo/kernels.cuh

```diff
@@ -264,13 +264,77 @@ __device__ OutT* alignPointer(InT input) {
     raft::alignTo(reinterpret_cast<size_t>(input), sizeof(OutT)));
 }
 
+// 32-bit FNV1a hash
+// Reference: http://www.isthe.com/chongo/tech/comp/fnv/index.html
+const uint32_t fnv1a32_prime = uint32_t(16777619);
+const uint32_t fnv1a32_basis = uint32_t(2166136261);
+
+DI uint32_t fnv1a32(uint32_t hash, uint32_t txt) {
+  hash ^= (txt >> 0) & 0xFF;
+  hash *= fnv1a32_prime;
+  hash ^= (txt >> 8) & 0xFF;
+  hash *= fnv1a32_prime;
+  hash ^= (txt >> 16) & 0xFF;
+  hash *= fnv1a32_prime;
+  hash ^= (txt >> 24) & 0xFF;
+  hash *= fnv1a32_prime;
+  return hash;
+}
```

Comment:
nitpick: could we maybe reuse [this logic](https://github.com/rapidsai/raft/blob/branch-0.18/cpp/include/raft/random/rng_impl.cuh#L236) from raft?

Reply:
We need 32-bit FNV1a here and raft's is 128-bit.

Reply:
oh.. then ignore my comment above. Thanks for the clarification.